### DISCUSSION TITLE
Is it possible to run the framework in an Severless AWS Lambda?

### DISCUSSION THREAD

**Discussion by ghost** [Netflix/dgs-framework#851] [Q&A] [answered]
Hi everyone.

Right now I am very happy using the framework, it's great but I want to know if anyone tried to run in an Serverless AWS Lambda environment?
I'm afraid of the cold starts because of Spring basically.

Another question is if it's possible to run the framework with another framework that is not Spring(Like micronaut, quarkus, etc).

Thank you!

**Comment by paulbakker** [comment, marked as answer]
DGS is tied to Spring Boot, it's not designed to be used with other frameworks. 

For cold starts - a Spring Boot/DGS application takes ~2 seconds to start up. If that's acceptable or not depends on your need of course. Additional startup time that you commonly see with Spring applications is caused by other libraries/frameworks such as ORM, grpc clients, etc. You will run into the same startup cost with other frameworks if using such things.
